src/Profiles.ts

```diff
@@ -1351,32 +906,29 @@ export class Profiles {
         const OrigProfileInfo = this.loadedProfile.profile;
         const NewProfileInfo = ProfileInfo.profile;
 
-        const profileArray = Object.keys(this.loadedProfile.profile);
+        const profileArray = Object.keys(NewProfileInfo);
         for (const value of profileArray) {
             if (value === "user" || value === "password") {
                 if (!rePrompt) {
-                        OrigProfileInfo.user = NewProfileInfo.user;
-                        OrigProfileInfo.password = NewProfileInfo.password;
+                    if (NewProfileInfo.user) { OrigProfileInfo.user = NewProfileInfo.user; }
+                    if (NewProfileInfo.password) { OrigProfileInfo.password = NewProfileInfo.password; }
```

Comment:
These if statements are not working correctly. After the Check Creds pop up box is checked and new are entered then the Save Creds pop up shows and save is clicked, nothing is being saved to yaml or session.  The next time I click on the profile the Check Creds immediately pops back up. May need to check for `!== undefined` or something

Reply:
just tested and this works
```
 if (NewProfileInfo.user !== undefined) { OrigProfileInfo.user = NewProfileInfo.user; }
 if (NewProfileInfo.password !== undefined) { OrigProfileInfo.password = NewProfileInfo.password; }
```

Reply:
pushed that fix

Reply:
HA! pushing that fix also fixed the DCO lol